File: src/main/java/io/dplabs/bookshelf/models/Category.java
package io.dplabs.bookshelf.models;

import lombok.Data;

import javax.persistence.Entity;
import javax.validation.constraints.NotBlank;

@Entity(name="categories")
@Data
public class Category extends BaseEntity {
    @NotBlank(message = "Name is mandatory")
    private String name;
}

File: backend/src/repository/entities/category.ts
import { Column, Entity, OneToMany } from 'typeorm';
import { BaseEntity } from './base-entity';
import { Book } from './book';

@Entity({ name: 'categories' })
export class Category extends BaseEntity {
  @Column({ unique: true })
  name: string;

  @OneToMany(() => Book, (book) => book.category)
  books: Book[];
}
